waku/waku_rln_relay/rln_relay.nim

```diff
@@ -445,17 +462,20 @@ proc mount(
   (await groupManager.startGroupSync()).isOkOr:
     return err("could not start the group sync: " & $error)
 
-  return ok(
-    WakuRLNRelay(
-      groupManager: groupManager,
-      nonceManager:
-        NonceManager.init(conf.rlnRelayUserMessageLimit, conf.rlnEpochSizeSec.float),
-      rlnEpochSizeSec: conf.rlnEpochSizeSec,
-      rlnMaxEpochGap: max(uint64(MaxClockGapSeconds / float64(conf.rlnEpochSizeSec)), 1),
-      onFatalErrorAction: conf.onFatalErrorAction,
-    )
+  wakuRlnRelay = WakuRLNRelay(
+    groupManager: groupManager,
+    nonceManager:
+      NonceManager.init(conf.rlnRelayUserMessageLimit, conf.rlnEpochSizeSec.float),
+    rlnEpochSizeSec: conf.rlnEpochSizeSec,
+    rlnMaxEpochGap: max(uint64(MaxClockGapSeconds / float64(conf.rlnEpochSizeSec)), 1),
+    onFatalErrorAction: conf.onFatalErrorAction,
   )
 
+  # Start epoch monitoring in the background
+  asyncSpawn monitorEpochs(wakuRlnRelay)
```

Comment:
Instead of performing a _launch-and-forget_ approach, it is better to get a reference of the async-task future handle so that it can be gracefully cancelled afterwards.
See for example:
https://github.com/waku-org/nwaku/blob/1b532e8ab9efec70b0d8b79ba65e145e2b3cb76b/waku/waku_archive/archive.nim#L256
https://github.com/waku-org/nwaku/blob/1b532e8ab9efec70b0d8b79ba65e145e2b3cb76b/waku/waku_archive/archive.nim#L264